src/hooks/users.ts

```diff
@@ -1,63 +1,159 @@
 import { useMemo } from "react";
-import { isLoaded } from "react-redux-firebase";
 
 import { User } from "types/User";
-import { ValidStoreAsKeys } from "types/Firestore";
 
-import { usersSelector, usersByIdSelector } from "utils/selectors";
+import { worldUsersSelector, worldUsersByIdSelector } from "utils/selectors";
 import { WithId } from "utils/id";
 
-import { useSelector } from "hooks/useSelector";
-import { useUserLastSeenThreshold } from "hooks/useUserLastSeenThreshold";
-import { useVenueId } from "hooks/useVenueId";
-import { useFirestoreConnect } from "hooks/useFirestoreConnect";
+import { useSelector } from "./useSelector";
+import { useUserLastSeenThreshold } from "./useUserLastSeenThreshold";
+import { useConnectCurrentVenueNG } from "./useConnectCurrentVenueNG";
+import { useVenueId } from "./useVenueId";
+import { useFirestoreConnect, isLoaded } from "./useFirestoreConnect";
 
-const useConnectVenueUsers = () => {
+export const useConnectWorldUsers = () => {
   const venueId = useVenueId();
 
-  // @debt refactor this + related code so as not to rely on using a shadowed 'storeAs' key
-  //   this should be something like `storeAs: "currentVenueUsers"` or similar
-  useFirestoreConnect(
+  const { isCurrentVenueLoaded, currentVenue } = useConnectCurrentVenueNG(
     venueId
-      ? {
-          collection: "users",
-          where: ["enteredVenueIds", "array-contains", venueId],
-          storeAs: "users" as ValidStoreAsKeys, // @debt super hacky, but we're consciously subverting our helper protections
-        }
-      : undefined
+  );
+
+  useFirestoreConnect(() => {
+    if (!isCurrentVenueLoaded || !currentVenue || !venueId) return [];
+
+    const relatedLocationIds = [currentVenue.id];
+
+    if (currentVenue.parentId) {
+      relatedLocationIds.push(currentVenue.parentId);
+    }
```

Comment:
This part still doesn't handle venues that are nested multiple times (eg. `A -> B -> C`). And it's not going to get sibling venues and similar.

Is there a reason why we aren't/can't use `useRelatedVenues` here?

Reply:
The issues arise when there are 10+ venues(as in WaySpace for ex.). When this happens, our useFirestoreConnect's `array-contains-any` doesn't work properly

Reply:
Can we use the ‘chunk’ pattern in that case? As has been used elsewhere in the codebase (I think for something chat related, private chats maybe?)

Reply:
At the very least I think we need to be able to get the ‘root parent id’ here, as most users will have had to of come through that to register/enter the platform (previously discussed edgecases aside).

How we go about finding/getting that here, I’m open to suggestions.

One thought that I had was to run a series of ‘direct’ ‘manual’ queries to retrieve the venues and ‘wall up the tree’ until the ‘root parent venue’ is found, then caching that for use in this array.

Reply:
The thing you are referring to is probably this chunk of code
![image](https://user-images.githubusercontent.com/63194656/105707850-3a7e2280-5f1c-11eb-92b2-212012c96122.png)

And I don't think it functions properly. It will only load 10 chat users, I think

Reply:
> At the very least I think we need to be able to get the ‘root parent id’ here, as most users will have had to of come through that to register/enter the platform (previously discussed edgecases aside).
> 
> How we go about finding/getting that here, I’m open to suggestions.
> 
> One thought that I had was to run a series of ‘direct’ ‘manual’ queries to retrieve the venues and ‘wall up the tree’ until the ‘root parent venue’ is found, then caching that for use in this array.

Okay, I will find a way to do it. Sounds reasonable

Reply:
> The thing you are referring to is probably this chunk of code

I think this is what I was referring to (which isn't about chats at all it seems, and also doesn't appear to be using `useFirestoreConnect` either):

https://github.com/sparkletown/sparkle/blob/932c5c6724fe6cb75596e0c45a03b139da6fa15d/src/components/organisms/Badges/Badges.tsx#L42-L51

I'm pretty certain we used to (maybe it has since been removed) have some similar code that was loading private chats/similar, and was manipulating the `storeAs` key to have an 'index' for each 'chunk'.